Adding deployment to DigitalOcean App Platform

chris-on-code opened the pull request:
### What does it do?

Added 1 file to the deployments section to deploy Strapi to DigitalOcean's new Platform as a Service (App Platform)

### Why is it needed?

We should show users how to quickly deploy to DigitalOcean's new offering

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/documentation?pullRequest=238) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/documentation?pullRequest=238) before we can accept your contribution.<br/><hr/>**Chris Sev** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/documentation?pullRequest=238) it.</sub>

derrickmehaffy commented on docs/developer-docs/latest/setup-deployment-guides/deployment/hosting-guides/digitalocean-app-platform.md:
```diff
@@ -0,0 +1,99 @@
+Prior to starting this guide, you should have created a [Strapi project](/developer-docs/latest/getting-started/quick-start.md). And have read through the [configuration](/developer-docs/latest/setup-deployment-guides/deployment.md#application-configuration) section.
+
+::: tip
+Strapi does have a [One-Click](/developer-docs/latest/setup-deployment-guides/installation/digitalocean-one-click.md) deployment option for DigitalOcean and can also be deployed to DigitalOcean Droplets.
```
can you add the backlink to the droplet guide also

chris-on-code replied:
added the backlinks

derrickmehaffy commented on docs/developer-docs/latest/setup-deployment-guides/deployment/hosting-guides/digitalocean-app-platform.md:
```diff
@@ -0,0 +1,99 @@
+App Platform lets you deploy your application directly from a [GitHub](https://github.com) repo. [Gitlab](https://gitlab.com) is also supported.
+
+### Step 1. Log in to your [DigitalOcean account](https://cloud.digitalocean.com/login).
+
```
Not required but do you want to add some screenshots? We can create a new assets folder in https://github.com/strapi/documentation/tree/main/docs/developer-docs/latest/assets for deployment assets. I believe we are planning to at a later date but it's up to you if you want to do it now.

chris-on-code replied:
theres some movement happening in the ui so i think screenshots may get outdated quickly. i can revisit after a month or so

derrickmehaffy replied:
No problem, we will table it for now. Interested to see the UI changes 👀

derrickmehaffy reviewed (changes requested):
Only other things that might be needed is to modify the existing DO Droplets to specify it's for droplets and maybe adding in info on the url key in the `server.js` but we can always add that reference later

derrickmehaffy commented:
Hey @chris-on-code can you make the following changes (you have your PR branch locked so I can't push to it)

File: `docs/developer-docs/latest/setup-deployment-guides/deployment.md`
Line starts at 69

```
<div>
	<InstallLink link="deployment/hosting-guides/digitalocean.html">
		<template #icon>
			<svg width="178" height="177" viewBox="0 0 178 177" xmlns="http://www.w3.org/2000/svg"><g fill="#fff" fill-rule="evenodd"><path d="M89 176.5v-34.2c36.2 0 64.3-35.9 50.4-74-5.1-14-16.4-25.3-30.5-30.4-38.1-13.8-74 14.2-74 50.4H.8C.8 30.6 56.6-14.4 117.1 4.5c26.4 8.3 47.5 29.3 55.7 55.7 18.9 60.5-26.1 116.3-83.8 116.3z" fill-rule="nonzero"></path><path d="M89.1 142.5H55v-34.1h34.1zM55 168.6H28.9v-26.1H55zM28.9 142.5H7v-21.9h21.9v21.9z"></path></g></svg>
		</template>
		<template #title>DigitalOcean Droplets</template>
		<template #description>
			Manual step by step guide for deploying on DigitalOcean Droplets
		</template>
	</InstallLink>
</div>

<div>
	<InstallLink link="deployment/hosting-guides/digitalocean-app-platform.html">
		<template #icon>
			<svg width="178" height="177" viewBox="0 0 178 177" xmlns="http://www.w3.org/2000/svg"><g fill="#fff" fill-rule="evenodd"><path d="M89 176.5v-34.2c36.2 0 64.3-35.9 50.4-74-5.1-14-16.4-25.3-30.5-30.4-38.1-13.8-74 14.2-74 50.4H.8C.8 30.6 56.6-14.4 117.1 4.5c26.4 8.3 47.5 29.3 55.7 55.7 18.9 60.5-26.1 116.3-83.8 116.3z" fill-rule="nonzero"></path><path d="M89.1 142.5H55v-34.1h34.1zM55 168.6H28.9v-26.1H55zM28.9 142.5H7v-21.9h21.9v21.9z"></path></g></svg>
		</template>
		<template #title>DigitalOcean App Platform</template>
		<template #description>
			Manual step by step guide for deploying on DigitalOcean App Platform
		</template>
	</InstallLink>
</div>
```

Then modify the `docs/developer-docs/latest/setup-deployment-guides/deployment/hosting-guides/digitalocean.md` header from `# DigitalOcean` to `# DigitalOcean Droplets`

---

@Mcastres once we merge this PR can you update your meta PR to add in the required SEO stuff.

@pwizla we probably will need to update the contributing guide to include SEO stuff.

pwizla commented on docs/developer-docs/latest/setup-deployment-guides/deployment/hosting-guides/digitalocean-app-platform.md:
```diff
@@ -0,0 +1,99 @@
+
+For prototype applications, you can choose the Basic plan. For applications that are expecting production traffic, you can choose the Pro plan. You will also see the pricing for your database that you have chosen.
+
+Choose your container size based on how much traffic you believe your app will have. It is a good practice to start on the smaller sizes, monitor the metrics of your app, and scale up as your app grows. App Platform allows us to scale vertically or horizontally with the click of a button.
```
```suggestion
Choose your container size based on how much traffic you believe your app will have. It is a good practice to start on the smaller sizes, monitor the metrics of your app, and scale up as your app grows. App Platform allows DigitalOcean to scale vertically or horizontally with the click of a button.
```

Suggesting this because Strapi documentation is usually written from Strapi's point of view ;-)

pwizla reviewed (changes requested):
Thanks for the contribution, @chris-on-code !

I second Derrick's request to update the `deployment.md` file, otherwise readers will only find your document by specifically searching for it.

Also, I suggested a tiny change, see inline comment.

chris-on-code commented:
ok ive got all the new changes in

- added link on the main `deployment.md` file
- changed of droplet docs to `# DigitalOcean Droplets`
- small verbiage change requested

pwizla reviewed (approved):
Thanks, LGTM!

chris-on-code commented:
oh sorry i missed the "Deployments" word on the `deployment.md` page. added a commit

derrickmehaffy reviewed (approved):
LGTM

pwizla commented:
Great! It's merged and will be live with the next documentation deployment, before the StrapiConf ;-)
